fem/bilininteg.hpp

```diff
@@ -2144,6 +2181,28 @@ class DiffusionIntegrator: public BilinearFormIntegrator
                                        ElementTransformation &Trans,
                                        DenseMatrix &elmat);
 
+   virtual void AssemblePatchMatrix(const int patch, Mesh *mesh,
+                                    SparseMatrix*& smat);
+
+   /** Called by AssemblePatchMatrix for sparse matrix assembly on a NURBS patch
+    with full 1D quadrature rules. */
+   void AssemblePatchMatrix_fullQuadrature(const int patch, Mesh *mesh,
+                                           SparseMatrix*& smat);
+
+   /** Called by AssemblePatchMatrix for sparse matrix assembly on a NURBS patch
+    with reduced 1D quadrature rules. */
+   void AssemblePatchMatrix_reducedQuadrature(const int patch, Mesh *mesh,
+                                              SparseMatrix*& smat);
+
+   /// TODO: remove this before merging PR?
+   void AssemblePatchMatrix_simpleButInefficient(const int patch,
```

Comment:
Initially, I thought it may be helpful to keep these inefficient versions, as they are simpler to read and understand, as a stepping stone to understanding the other optimized integrators (which are rather cryptic). I have removed them now, because I suppose they have too much code to keep just for educational purposes.